app/controllers/concerns/api/users_controller.rb

```diff
@@ -21,5 +21,28 @@ def show
     def me
       render :show
     end
+
+    def suspend
```

Comment:
Mimics what is performed in the admin dashboard in the [`user_status` action](https://github.com/forem/forem/blob/main/app/controllers/admin/users_controller.rb#L118), which includes both:

- Make use of `Moderator::ManageActivityAndRoles.handle_user_roles` service class
- Create a Note that documents this action

Reply:
[Suggestion] This is probably out of the scope of this PR, but it strikes me as a anti pattern that we have to manually call `Note.create` after `Moderator::ManageActivityAndRoles.handle_user_roles`. Is there ever a case when we'd call `Moderator::ManageActivityAndRoles.handle_user_roles` with `user_status: "Suspected"` **without** a note? Should it require a note?

Reply:
I think this is a good suggestion and questions. Looking into this class a bit more thoroughly I see it already has a method to create a note ([here](https://github.com/forem/forem/blob/main/app/services/moderator/manage_activity_and_roles.rb#L51)), so I think I can come up with a small refactor to reuse this. I'll push the changes soon 👍🏼 

This service class is used indirectly (via inheritance of other classes) in a couple more places so I want to be careful though 

<img width="412" alt="Screen Shot 2022-07-13 at 08 45 22" src="https://user-images.githubusercontent.com/6045239/178763429-699d608b-4319-4b9f-8340-d4388d1c4636.png">

Reply:
Turns out I was a little confused by how the admin controller was creating a Note on its own 😓 

Since we're passing `user_params[:note_for_current_role]` parameter to the service class we don't need to create the Note, it will create one on its own ([see here](https://github.com/forem/forem/blob/main/app/services/moderator/manage_activity_and_roles.rb#L137)). With the previous implementation (before these last few commits) we were actually left with 2 new Notes.

To corroborate this works as expected I [updated the spec a bit](https://github.com/forem/forem/pull/18043/files#diff-8ec6f3422418fdc89fb9da85ce6384e3f40e5a8dd868aec86e1c15c77670dd27R180-R192) to ensure this 🙂